src/components/input/input.js

```diff
@@ -650,7 +651,21 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       // Using the $validators for triggering the update works very well.
       renderCharCount();
 
-      return ( modelValue || element.val() || viewValue || '' ).length <= maxlength;
+      // Force the value into a string since it may be a number,
+      // which does not have a length property.
+      return String(modelValue || element.val() || viewValue || '').length <= maxlength;
+    };
+
+    ngModelCtrl.$validators['required'] = function(modelValue, viewValue) {
+      if (!attr.required) {
+        return true;
+      }
+
+      var elementVal = modelValue || element.val() || viewValue || '';
+      elementVal = ngTrim && angular.isString(elementVal) ? elementVal.trim() : elementVal;
+      // Force the value into a string since it may be a number,
+      // which does not have a length property.
+      return angular.isDefined(elementVal) && String(elementVal).length;
```

Comment:
Nit: At this point `elementVal` can't be undefined. At worse, it'll be the empty string :grin:

Reply:
I was trying to handle the `0` case, but I can see from your comments above that this isn't the place to do it. Thanks.